File: api/src/main/java/com/sirius/domain/model/db/UserDTO.java
package com.sirius.domain.model.db;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sirius.domain.controller.db.UserRolesController;
import com.sirius.exception.ResourceNotFoundException;
/*import org.hibernate.validator.constraints.NotBlank;*/
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.util.Arrays;
import java.util.Date;
import java.util.List;
import java.util.stream.Collectors;

@Entity
@Table(name = "user")
@EntityListeners(AuditingEntityListener.class)
public class UserDTO {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotBlank
    private String username;

    @NotBlank
    @JsonIgnore
    private String password;

    private String firstName;

    private String lastName;

    private String email;

    private String gender;

    private String ipAddress;
    
    private Date registered;

    private Date lastLogin;

    private Boolean isOnline;

    private String picture;

    private String userType;

    private Boolean verified;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPasswordHash() {
        return password;
    }

    public void setPasswordHash(String password) {
        this.password = password;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getIpAddress() {
        return ipAddress;
    }

    public void setIpAddress(String ipAddress) {
        this.ipAddress = ipAddress;
    }

    public Date getRegistered() {
        return registered;
    }

    public void setRegistered(Date registered) {
        this.registered = registered;
    }

    public Date getLastLogin() {
        return lastLogin;
    }

    public void setLastLogin(Date lastLogin) {
        this.lastLogin = lastLogin;
    }

    public Boolean getIsOnline() {
        return isOnline;
    }

    public void setIsOnline(Boolean isOnline) { this.isOnline = isOnline; }

    public String getPicture() {
        return picture;
    }

    public void setPicture(String picture) {
        this.picture = picture;
    }

    public String getUserType() {
        return userType;
    }

    public void setUserType(String userType) {
        this.userType = userType;
    }

    public Boolean getVerified() {
        return verified;
    }

    public void setVerified(Boolean verified) {
        this.verified = verified;
    }

    public UserDTO orElseThrow(ResourceNotFoundException exception) {
        throw exception;
    }
}

File: web/sirius_flutter/lib/views/user/UserDTO.dart
class UserDTO {
  int id;

  String username;

  String firstName;

  String lastName;

  String email;

  String gender;

  int registered;

  int lastLogin;

  bool isOnline;

  String picture;

  UserDTO(
      {this.id,
        this.username,
        this.firstName,
        this.lastName,
        this.email,
        this.gender,
        this.registered,
        this.lastLogin,
        this.isOnline,
        this.picture,});

  factory UserDTO.fromJson(Map<String, dynamic> json) {
    return UserDTO(
      id: json['id'],
      username: json['username'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      gender: json['gender'],
      registered: json['registered'],
      lastLogin: json['lastLogin'],
      isOnline: json['isOnline'],
      picture: json['picture'],
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['username'] = this.username;
    data['firstName'] = this.firstName;
    data['lastName'] = this.lastName;
    data['email'] = this.email;
    data['gender'] = this.gender;
    data['registered'] = this.registered;
    data['lastLogin'] = this.lastLogin;
    data['isOnline'] = this.isOnline;
    data['picture'] = this.picture;
    return data;
  }

}
